corehq/apps/users/models.py

```diff
@@ -1764,17 +1764,21 @@ def get_language_code(self):
     def location(self):
         from corehq.apps.locations.models import Location
         if self.location_id:
-            return Location.get(self.location_id)
-        else:
-            return None
+            try:
+                return Location.get(self.location_id)
+            except ResourceNotFound:
+                self.unset_location()
```

Comment:
Fine with it returning `None` instead of failing hard.

What are you saying happens now if a location is archived? Does that document get deleted from couch or what would happen to this call? Agree that calling `unset_location` on relevant users would be a good thing to do either way.